lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -1711,19 +1692,19 @@ class TopologicalSort {
   // the entry itslef.
   void visit(SPIRVEntry *E) {
     DFSState &State = EntryStateMap[E];
+    if (E->getOpCode() == OpTypePointer) {
+      SPIRVTypePointer *Ptr = static_cast<SPIRVTypePointer *>(E);
+      if (EntryStateMap[Ptr->getElementType()] == Discovered) {
```

Comment:
May be it worth to add a comment, why do we insert OpTypeForwardPointer pointer if we found the same OpTypePointer in the module.

Reply:
Done